assets/tests/abac_tests.yaml

```diff
@@ -530,7 +530,59 @@ tests:
                 relation: viewer
                 user: user:jon
             expectation: true
-  - name: direct_relations_with_condition
+        listObjectsAssertions:
+          - request:
+              user: user:jon
+              type: document
+              relation: viewer
+            errorCode: 2000
+          - request:
+              user: user:jon
+              type: document
+              relation: viewer
+            context:
+              s: "hello"
+            errorCode: 2000
```

Comment:
This should be `document:1` not an error, yeah?

Reply:
I thought this too, but... ListObjects is supposed to return an error only if it cannot fulfil `max_results`. In all these tests, `maxResults = 1000`, but we never expect that many results, so we return an error.

We'd have to change ReverseExpansion quite a bit to ignore errors for objects that it has already sent.

Reply:
Because there are two conditions in the model, only one which has enough context to evaluate true, when the graph expands and evaluates `folder with xcond`, there will be an error. As Maria notes above, this _should_ depend on the number of max results but here we are far from the max results.

Reply:
Created an issue to track this work separately.
https://github.com/openfga/openfga/issues/1511